### PR TITLE
bugfix - maintain subscription comment count on cache clear

### LINKED ISSUE #3515: New comment count prunes subscriptions prematurely
http://www.reddit.com/r/resissues/comments/56ew0h/_/

When cleaning comments, if subscription length option is higher than clean comments option, clean comments should not delete entries that are actively subscribed to.

### PR DESCRIPTION
**Pull request by ssonal**
Should fix #3515

### CODE REVIEW

**Comment by andytuba** [comment]
Nice turnaround ☺️ this will certainly solve the problem.  It would be neat to see an iteration that does clean out items which aren't subscribed, but this definitely helps.

**Comment by ssonal** [comment]
I wanted to get this done with minimal effort because I'm not very familiar with the codebase so I didn't want to risk breaking stuff elsewhere.

**Comment by andytuba** [comment]
Nice turnaround! It would be neat to specifically clean out comments that arent actively subscribed, instead of skipping the cleaning entirely.

**Comment by andytuba** [comment]
Wait sorry, I think k you did exactly that and I need to read closer..

File: lib/modules/newCommentCount.js [hunk 1]
```diff
@@ -385,8 +385,13 @@ function updateCommentCount(mycomment) {
 async function cleanCache() {
 	await getLatestCommentCounts();
 	const now = Date.now();
+	const skip = module.options.cleanComments.value < module.options.subscriptionLength.value;
```
Comment (erikdesjardins) [inline lib/modules/newCommentCount.js]: This is correct, but I think we should leave this out and ignore all comment counts where `subscriptionDate` is defined, regardless of `subscriptionLength`. That way the logic for removing thread subscriptions and removing comment counts are kept separate.

File: lib/modules/newCommentCount.js [hunk 2]
```diff
@@ -389,7 +389,11 @@ async function cleanCache() {
 	for (const i in commentCounts) {
 		if (commentCounts[i] && ((now - commentCounts[i].updateTime) > (86400000 * module.options.cleanComments.value))) {
 			// Do not automatically delete comments belonging to actively subscribed threads
-			if (skip && typeof commentCounts[i].subscriptionDate !== 'undefined') {
+			if (skip) {
+				// Delete expired subscriptions
+				if (typeof commentCounts[i].subscriptionDate === 'undefined') {
```
Comment (andytuba) [inline lib/modules/newCommentCount.js]: subscriptions should be cleaned per the subscription length option, comment threads that aren't subscribed should be cleared per the clean comments option. More specifically it looks like Erik has the good ideas.

File: lib/modules/newCommentCount.js (hunk 1 above)
Comment (ssonal) [inline lib/modules/newCommentCount.js]: Makes sense. Good way to avoid bugs from future changes.

Comment (ssonal) [inline lib/modules/newCommentCount.js]: I think the latest commit ought to do it.

**Pull request merged**